module/apmawssdkgo/session.go

```diff
@@ -52,6 +98,38 @@ func WrapSession(s *session.Session) *session.Session {
 	return s
 }
 
+type config struct {
+	tracer   *apm.Tracer
+	registry *prometheus.Registry
+}
+
+// Option sets options for tracing server requests.
+type Option func(*config)
+
+// WithTracer returns a Option which sets t as the tracer to use for tracing
+// requests made with the AWS SDK.
+func WithTracer(t *apm.Tracer) Option {
+	if t == nil {
+		panic("t == nil")
+	}
+	return func(c *config) {
+		c.tracer = t
+	}
+}
+
+// WithRegistry returns an Option which sets r as the prometheus.Registry to
+// use when recording AWS SDK metrics. If no registry is supplied, a new one
+// will be instantiated.Metrics will be reported to the APM server regardless
+// of the prometheus.Registry supplied.
+func WithRegistry(r *prometheus.Registry) Option {
```

Comment:
What's the reason for making the registry configurable? Seems to me `prometheus/client_golang` is an implementation detail. In theory we might later want to specify units for metrics, and that won't be possible as long as we use Prometheus. I also tend to think we should keep the scope minimal, and only support sending to APM Server.

Reply:
hm, I guess I didn't really have any reasoning other than "maybe people want to pass in their registry". But it's simpler from an API perspective, and if folks for some reason double register, prevents a panic when trying to register the same metrics twice.